.gitignore

```diff
@@ -1,39 +1 @@
-# See http://help.github.com/ignore-files/ for more about ignoring files.
-
-# compiled output
-/dist
-/tmp
-/out-tsc
-
-# dependencies
-/node_modules
-
-# IDEs and editors
-/.idea
```

Comment:
Can we leav some of the IDEs? I personally use intellij and any attempt of working with this repo will end up with such an entry in this file

Reply:
If you are using IntelliJ you should add this to your global gitignore file

Reply:
I'm not convinced. Is that the best practice currently? Sounds too simple for anyone who does not have a (relevant) global `.gitignore` to accidentally commit unwanted files

Reply:
https://dev.to/oleksiyrudenko/please-do-not-pollute-projects-gitignore-3h8e

Reply:
I agree - we won't be able to enforce such a restriction on every contributor and I wouldn't like to have that in mind on every PR. Let's have a setup that will prevent unwanted content for common cases.

Reply:
What's most common? Some time ago it was atom, now it's vscode, IDEs come and go. I do not want to maintain that list. Everybody should keep their own shit away from the repo. People that are not aware of `~/.gitignore` will never learn to set it up properly if we do the work for them. But I do not want to lose more time discussing that. If you insist, I will restore the original .gitignore, but I will not maintain it regarding IDE/OS specific stuff.

Reply:
I do see the merit here. Consider that once you set up a global .gitignore for your specific IDE files, it will also save you the time tom maintain the ignores in each individual repo you shall have on your machine. win-win

Reply:
I personally wasn't aware of global gitignore and I also practiced project-based gitignore files.
I also acknowledge your arguments - I may agree with them. Still, it will be generally my duty to filter unwanted content from the repo, thus I'd like to have some entries related to some IDE's (VS Code, IntelliJ, etc.) hoping it will spare some time in future for me.
Thanks for restoring it!